src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,185 @@
+        elif iver_num > sver_num:
+            return 1
+
+    return 0
+
+
+def _check_daqmx_is_installed():
+    """
+    Check for existing installation of DAQmx by checking for Version information.
+    Version information is read from the registry key.
+    """
+    try:
+        with winreg.OpenKeyEx(
+            winreg.HKEY_LOCAL_MACHINE,
+            r"SOFTWARE\National Instruments\NI-DAQmx\CurrentVersion",
+            0,
+            winreg.KEY_READ | winreg.KEY_WOW64_32KEY,
+        ) as daqmx_reg_key:
+            product_name = winreg.QueryValueEx(daqmx_reg_key, "ProductName")[0]
+            product_version = winreg.QueryValueEx(daqmx_reg_key, "Version")[0]
+
+        if product_name == "NI-DAQmx":
+            return product_version
+    except Exception as e:
+        print(f"Info: No existing NI-DAQmx installation found")
+
+    return False
+
+
+def _get_temp_file_name():
+    """
+    Generate a temporary file name.
+    """
+    try:
+        _, temp_file_path = tempfile.mkstemp()
+        return temp_file_path + "_daq-mx-online.exe"
+    except Exception as e:
+        print(f"error: Unable to create temporary file: {e}")
+        return None
+
+
+def _get_url_and_version_for_windows_installer(metadata_url):
+    """
+    Parse the JSON data and retrieve the download link and version information.
+    """
+    print("Log1")
+    try:
+        with urllib.request.urlopen(metadata_url) as url:
+            data = json.load(url)
+            windows_os_meta = data.get("Windows", [])
+
+        for metadata in windows_os_meta:
+            location = metadata.get("Location")
+            version = metadata.get("Version")
+            if location and version:
+                return location, version
+
+    except Exception as e:
+        print(f"Unable to get download url: {e}")
+    return None, None
+
+
+def _install_daqmx(download_url, temp_file):
+    """
+    Download and launch NI-DAQmx Driver installation in an interactive mode
+    """
+    try:
+        urllib.request.urlretrieve(download_url, temp_file)
+        subprocess.Popen([temp_file], shell=True)
+    except Exception as e:
+        print(f"Error installing NI-DAQmx: {e}")
+
+
+def _ask_user_confirmation(question):
+    """
+    Prompt for user confirmation
+    """
+    while True:
+        response = input(question + " (yes/no): ").strip().lower()
+        if response in ["yes", "y"]:
+            return True
+        elif response in ["no", "n"]:
+            return False
+        else:
+            print("Please enter 'yes' or 'no'.")
+
+
+def install_daqmx_software():
+    """
+    Get the platform name on which the Python script is being executed.
+    Check for the existing installaion of NI-DAQmx Driver.
+    If NI-DAQmx driver is already installed it will prompt for an upgrade if applicable.
+    Download and launch NI-DAQmx Driver installation in an interactive mode.
+    """
+
+    if sys.platform.startswith("win"):
+        if _is_admin():
```

Comment:
Is this admin check required on Windows? Can the installer elevate to administrator when you run it? Test with a standard user account on a machine that has UAC enabled.

If the user is using a standard user account, running Python scripts as administrator may cause them problems down the line, such as creating cache files with administrator permissions that the standard user can't delete or overwrite.

Reply:
@bkeryan , Installer can elevate to administrator and I have tested that.  We need administrator permissions to read windows registry, I will check with the product owner and/or @zhindes may be we can drop checking the existing installation of DAQmx software and we dont need the admin permissions.

Reply:
> We need administrator permissions to read windows registry,

You should not need administrator permissions to read that registry key. Please try it again with your current code. Perhaps you were specifying `access=winreg.KEY_WOW64_32KEY` without `winreg.KEY_READ` at one point?

![image](https://github.com/ni/nidaqmx-python/assets/17914399/d861c439-f438-41a6-b7a2-54bf3f55d954)

Reply:
Yep, on Windows I expect no admin privileges will be required. For Linux that is probably not the case, but we'll think about that later. Glad you've confirmed.